local/client/conn/conn.go

```diff
@@ -0,0 +1,66 @@
+package conn
+
+import (
+	"github.com/scootdev/scoot/local/protocol"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"log"
+	"net"
+	"time"
+)
+
+// A Dialer can dial a connection to the Scoot server.
+// It's useful to have this as a separate interface so you can wait to
+// connect until you need the connection. This allows clients to do client-side
+// only operations (e.g., printing help) without erroring if the server is down.
+type Dialer interface {
+	Dial() (Conn, error)
+}
+
+type Conn interface {
+	Echo(arg string) (string, error)
+	Close() error
+}
+
+func UnixDialer() (Dialer, error) {
+	socketPath, err := protocol.LocateSocket()
+	if err != nil {
+		return nil, err
+	}
+	return &dialer{socketPath}, nil
+}
+
+type dialer struct {
+	socketPath string
+}
+
+func dial(addr string, timeout time.Duration) (net.Conn, error) {
+	return net.DialTimeout("unix", addr, timeout)
+}
+
+func (d *dialer) Dial() (Conn, error) {
+	log.Println("Dialing ", d.socketPath)
+	c, err := grpc.Dial(d.socketPath, grpc.WithInsecure(), grpc.WithDialer(dial))
```

Comment:
Oh, I see. I could move func dial to be an anonymous func inside Dial(). I tend to prefer explicit small func's and prefer to leave anonymous inner func's for when they're operating as a closure and capturing state.

WDYT?